src/lib/warn-upload-overwrite.js

```diff
@@ -0,0 +1,73 @@
+const fs = require('./sync-fs');
+const jsonDiff = require('json-diff');
+const readline = require('readline-sync');
+
+const question = 'You are trying to modify a configuration that has been modified since your last upload. Do you want to?';
+const responseChoicesWithoutDiff = [
+  'Overwrite the changes', 
+  'Abort so that you can update the configuration'
+];
+const responseChoicesWithDiff = responseChoicesWithoutDiff.concat([ 'View diff' ]);
+
+const preUpload = async (projectDir, db, localDoc) => {
+
+  // Pull remote _rev
+  let remoteRev;
+  let remoteDoc;
+  try {
+    remoteDoc = await db.get(localDoc._id);
+    remoteRev = remoteDoc._rev;
+  } catch (e) {
+    // continue regardless of error
+  }
+
+  // Pull local _rev
+  let localRev;
+  try {
+    localRev = fs.read(`${projectDir}/._revs/${localDoc._id}`);
+  } catch (e) {
+    // continue regardless of error
+  }
+
+  // Compare _revs
+  // If _revs are different, show prompt
+  if (!localRev) {
+    if (!readline.keyInYN(`We can't determine if you're going to overwrite someone else's changes or not. Would you like to proceed?`)) {
+      throw new Error('configuration modified');
+    }
+  } else if (localRev !== remoteRev) {
+    let diff; 
+    if (localDoc.settings && remoteDoc.settings) {
+      diff = jsonDiff.diffString(remoteDoc.settings, localDoc.settings);
+    } else {
+      diff = jsonDiff.diffString(remoteDoc, localDoc);
```

Comment:
I think this will always fail, because the localDoc won't have the same _rev as the remoteDoc, right? For example, when uploading forms, the `localDoc` is a new object with `_rev` undefined. So we need to do a compare ignoring the _revs.

Also, I'm not sure if comparing like this will work for attachments - they often look different on local docs vs remote docs. We might have to do some custom comparison to check the `md5` hash of the attachments rather than just the json.